flux-refineck/src/checker.rs

```diff
@@ -923,6 +927,37 @@ impl<'a, 'tcx, P: Phase> Checker<'a, 'tcx, P> {
     }
 }
 
+fn type_of_shift<'a>(
```

Comment:
yikes, I didn't know shift operators could be used with mixed types. We need a more general infrastructure to declare types for operators, but this will do for now.

Reply:
Yeah, it makes sense to me to allow `i32 << u32` but i don't really understand why you would allow a signed shift operand.